### ISSUE TITLE
Would it be possible to switch gem ownership?

### ISSUE THREAD

**Issue by bkuhlmann** [rubyworks/xdg#6]
## Overview

It doesn't look like this project is maintained anymore and was wondering if I could take ownership of the Ruby Gems namespace for deploying new versions of the XDG library? This project could remain as is but I'd end up publishing a new XDG project with the ability to publish newer versions of the XDG library to Ruby Gems. Would the maintainers of this project be willing to do that?

## Details

I maintain the [Runcom](https://github.com/bkuhlmann/runcom) project which uses XDG under the covers as well as provides extended functionality for application developers. While the project currently doesn't support *all* of the XDG capabilities, I'm working on a [feature branch](https://github.com/bkuhlmann/runcom/compare/release) which *will* provide support for *all* XDG environment variables. I plan to release this at the end of this year or first of the new year as soon as Ruby 2.6 is released. :tada: My version of XDG does use a different API but the functionality that people need won't change.

## Next Actions

If any of this sounds good, please comment here and we can figure out a path forward. Thanks!

**Comment by peterhoeg** [comment]
@trans, any thoughts on passing the torch to an active maintainer?

**Comment by trans** [comment]
Sorry for not being more present. Life, Job ... blah blah blah...

But yes, transferring to a new maintainer is fine. My only worry is that I wouldn't want to see the gem just replaced with a totally redesigned API all of a sudden like. The code might be dusty but it's a simple library, and if any other projects out there are using it -- at this point I imagine in a largely "done and forgotten" manner -- I would hate to break their code and have them have to revisit and fix. That said, if a sufficient argument can be made for why the API should change, a well managed transition to a new API would of course be fine.

**Comment by bkuhlmann** [comment]
@trans No worries, glad I could get a hold of you! :wink: ...and I definitely sympathize with a busy life.

One way we could tackle this transition is by doing something like the following:

- Release a `2.2.0` version of *your* gem where the only change is an update to the project README and `gemspec` that includes a *strong* warning message upon gem install/upgrade where everyone is notified that in the 3.0.0 version, there will be breaking API changes.
- Release a `3.0.0` version which then utilizes the API changes I have in the `Runcom` gem.

The nice thing about this approach is those who are using pessimistic locking in their Gemfile will have to manually *choose* to upgrade to 3.0.0. Those who don't will get a bit of a shock but should at least know something is up prior to this with the `2.2.0` release. Regardless, for those who don't want to move to `3.0.0`, then they can simple lock their Gemfile at `2.2.0` until they wish to upgrade.

I can also help with the changes that would be needed in this project if you like. I can submit PRs with the changes suggested above. If that's too much, you could make me a co-owner of this project and I could work out the details that way too.

I'd still need to coordinate and/or be given access to RubyGems for this gem, ultimately. In order for me to publish *new* versions of the `xdg` gem, I'd need to have access to the RubyGems namespace. You could do that now by adding my email (`brooke@alchemists.io`) [as an owner](https://guides.rubygems.org/command-reference/#gem-owner).
 
Would that be acceptable?

**Comment by bkuhlmann** [comment]
@trans Wanted to check back in...any thoughts on the above? ↑ If you were to grant me *temporary* ownership of this project, I could help you move things forward via pull requests and/or direct changes based on how much time you have to review. I can also fork this project and issue pull requests the standard way too. Either way is fine by me but want to be sensitive to how much availability you have as I'm happy to do the heavy lifting. :wink:

**Comment by bkuhlmann** [comment]
@trans Wanted to check back in again. It's been a while and wanted to touch base. Any updates on this? Can I help in any way?

**Comment by bkuhlmann** [comment]
In an effort to be transparent. I've been granted access to the RubyGems XDG namespace space and released `2.2.4` version which has the following post-install message:

> ‼️ Development of this project has been transfered here: https://github.com/bkuhlmann/xdg. This will be the last version supported within the 2.2.x series. Be aware major breaking changes will be introduced in the upcoming 3.0.0 release (2019-06-01).

This will help anyone still using the old 2.2.x series to prep for the upcoming 3.0.0 release. :tada:

**Comment by bkuhlmann** [comment]
:information_source: Closing this issue as I'll proceed with future updates via the new [XDG](https://github.com/bkuhlmann/xdg) project.

**Comment by ryanmt** [comment]
2.2.4 fails to install w/ a missing file. 

For instance:
`~/.rbenv/versions/2.5.1/lib/ruby/gems/2.5.0/gems/xdg-2.2.4/lib/xdg.rb:28:in `require_relative': cannot load such file -- ~/.rbenv/versions/2.5.1/lib/ruby/gems/2.5.0/gems/xdg-2.2.4/lib/xdg/index (LoadError)`

FYI.

**Comment by bkuhlmann** [comment]
@ryanmt What version of Ruby are you using? I'm using Ruby 2.6.3 and not seeing any issues. It might be possible this was a pre-existing issue? In case it helps, the *only* difference between the 2.2.3 and 2.2.4 release is an update to the version and post install message within the `Index.yml`. No other changes were applied.

**Comment by ryanmt** [comment]
I'm secondary to this, but the installer is reporting that 2.2.4 is causing issues.  Let me confirm the behavior myself first. Thanks for the prompt response. 

I can confirm that requiring `xdg` in irb, as well as any derivative gem installations are broken w/ 2.2.4.

**Comment by ryanmt** [comment]
I would suspect ` Add dynamic index metadata lookup via constants.` is related.

diff-ing the install dirs gives me: 
```
Common subdirectories: /home/<user>/.gem/gems/xdg-2.2.3/demo and /home/<user>/.gem/gems/xdg-2.2.4/demo
diff /home/<user>/.gem/gems/xdg-2.2.3/HISTORY.md /home/<user>/.gem/gems/xdg-2.2.4/HISTORY.md
2a3,10
> ## 2.2.4 / ?
> 
> 
> Changes:
> 
> * Add dynamic index metadata lookup via constants.
> 
> 
Common subdirectories: /home/<user>/.gem/gems/xdg-2.2.3/lib and /home/<user>/.gem/gems/xdg-2.2.4/lib
Common subdirectories: /home/<user>/.gem/gems/xdg-2.2.3/test and /home/<user>/.gem/gems/xdg-2.2.4/test
```

**Comment by ryanmt** [comment]
@bkuhlmann Sorry I missed this question earlier, but this is linux and ruby 2.6.2 right now.

**Comment by bkuhlmann** [comment]
🙇 @ryanmt. I'm digging into this now. My guess is when I packaged the gem, I packaged it off of `master` assuming that `2.2.3`. I'm going to debug what you've posted above and what @chrisjohnson ran into as well.

If you don't mind, I'm would encourage you to follow the discussion here: https://github.com/bkuhlmann/xdg/issues/1. I say this because I'm actively working to move everything over to that project.
